### ISSUE TITLE
Model wrapper

### ISSUE THREAD

**Issue by JLrumberger** [Kainmueller-Lab/Pathology-Foundation-Model-Benchmark#9]
### Instructions
Write a model wrapper that can be used for each of the foundation model backbones. The simple version should 
1. Initialize the different foundation models and their preprocessing function based on a name string
2. Add a trainable head on top for dense prediction tasks: [simple version: resize + softmax layer](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/DINOv2/Train_a_linear_classifier_on_top_of_DINOv2_for_semantic_segmentation.ipynb)

**Design overview**
1. SimpleSegmentationModel class that has the foundation model encoder as .model and the associated preprocessing as .transform and adds a single softmax layer + bilinear resizing on top
2. A function that takes a string indicating the foundation model and returns the model and weights.

**Code mockup**

```python
import torch
import timm
from timm.data.transforms_factory import create_transform
from timm.data import resolve_data_config
from huggingface_hub import login


class SimpleSegmentationModel(torch.nn.Module):
    """Simple segmentation model that uses a backbone and a linear head."""
    def __init__(self, model_name, num_classes):
        super(SimpleSegmentationModel, self).__init__()
        self.model, self.transform, model_dim = load_model_and_transform(model_name)
        self.head = torch.nn.Linear(model_dim, num_classes, requires_grad=True)
        self.model.eval()
        self.freeze_backbone()

    def freeze_backbone(self):
        """Freeze the backbone of the model."""
        for param in self.model.parameters():
            param.requires_grad = False

    def unfreeze_backbone(self):
        """Unfreeze the backbone of the model."""
        for param in self.model.parameters():
            param.requires_grad = True

    def forward(self, x):
        """Forward pass of the model.
        
        Args:
            x: Input tensor of shape (b, c, h, w)
        Returns:
            logits: Output logits of shape (b, num_classes, h, w)
        """
        b,c,h,w = x.shape
        x = self.transform(x)
        patch_embeddings = self.model(x) # 1, 14, 14, 1024
        logits= self.head(patch_embeddings)
        logits = torch.nn.functional.interpolate(
            logits, size=(h,w), mode="bilinear", align_corners=False
        )
        return logits


def load_model_and_transform(model_name):
    if model_name == "UNI":
        login()
        model = timm.create_model(
            "hf-hub:MahmoodLab/UNI", pretrained=True, init_values=1e-5, dynamic_img_size=True
        )
        model.forward = lambda x: model.forward_features(x)[:,1:,:].reshape(x.shape[0], 14,14, -1)
        transform = create_transform(**resolve_data_config(model.pretrained_cfg, model=model))
        with torch.no_grad():
            model_dim = model(torch.zeros(1,3,224,224)).shape[-1]
    elif model_name == "XYZ":
        NotImplementedError("Model not implemented")
    return model, transform, model_dim
```

**Required inputs**

Some H&E pathology image or batch, foundation model checkpoints and preprocessing functions.

**Output files**

None

**Timeline**
Give a rough estimate for how long you think the project will take. In general, it's better to be too conservative rather than too optimistic. 
- [x] A couple days
- [ ] A week
- [ ] Multiple weeks. For large projects, make sure to agree on a plan that isn't just a single monster PR at the end. 

Estimated date when a fully implemented version will be ready for review:

Estimated date when the finalized project will be merged in:
